vsintegration/src/FSharp.Editor/CodeLens/AbstractCodeLensDisplayService.fs

```diff
@@ -299,6 +295,54 @@ type CodeLensDisplayService (view : IWpfTextView, buffer : ITextBuffer, layerNam
             ignore e
 #endif
 
-    abstract LayoutUIElementOnLine : IWpfTextView -> ITextViewLine -> Grid -> unit
-
-    abstract AsyncCustomLayoutOperation : int Set -> ITextSnapshot -> unit Async
\ No newline at end of file
+    /// Layouts all stack panels on the line
+    member self.LayoutUIElementOnLine _ (line:ITextViewLine) (ui:Grid) =
+        let left, top = 
+            try
+                let bounds = line.GetCharacterBounds(line.Start)
+                line.TextRight + 5.0, bounds.Top - 1.
+            with e ->
+#if DEBUG
+                logExceptionWithContext (e, "Error in layout ui element on line")
+#else
+                ignore e
+#endif
+                Canvas.GetLeft ui, Canvas.GetTop ui
+        Canvas.SetLeft(ui, left)
+        Canvas.SetTop(ui, top)
+    
+    member self.AsyncCustomLayoutOperation visibleLineNumbers buffer =
+        asyncMaybe {
+            // Suspend 5 ms, instantly applying the layout to the adornment elements isn't needed 
+            // and would consume too much performance
+            do! Async.Sleep(5) |> liftAsync // Skip at least one frames
+            do! Async.SwitchToContext self.UiContext |> liftAsync
+            let layer = self.CodeLensLayer
+            do! Async.Sleep(495) |> liftAsync
```

Comment:
Can you please make all such delays configurable via VS F# settings?

Reply:
Yep, good suggestion. I hope we'll get rid of them though, hardcoded timeouts are usually an antipattern